src/parseWithPointers.ts

```diff
@@ -1,445 +1,205 @@
-/*
-Right now the parseWithPointers function doesn't make use of parse function exposed by jsonc-parser.
-The reason we had to fall back to tokenizer is quite simple - lines and columns.
-jsonc-parser doesn't expose any onNewLine hook, so the only piece of information we are given is offset and length,
-which are not sufficient to build the list of pointers.
-That's why we had to bring the parse function in and modify it to let us have the pointers with correct location.
+import { DiagnosticSeverity, IDiagnostic } from '@stoplight/types/diagnostics';
+import { IParserASTResult, IParserResult } from '@stoplight/types/parsers';
+import { JSONVisitor, Node, NodeType, ParseErrorCode, ParseOptions, printParseErrorCode, visit } from 'jsonc-parser';
+import { IJsonASTNode } from './types';
 
-We should get rid of the current implementation of it. There are 2 ways to accomplish this:
-- either use offsets + lengths provided by jsonc-parser (would require changes to yaml/markdown parsers, related story SL-1958)
-or
-- create a PR with onNewLine hook
-
-Related task to address the issue: SL-2039
- */
-import { IParserResult, SourceMapParser } from '@stoplight/types/parsers';
-import { IValidationResult, ValidationSeverity, ValidationSeverityLabel } from '@stoplight/types/validations';
-import {
-  createScanner,
-  JSONPath,
-  JSONVisitor,
-  ParseErrorCode,
-  ParseOptions,
-  printParseErrorCode,
-  ScanError,
-  Segment,
-  SyntaxKind,
-} from 'jsonc-parser';
-import { encodePointerFragment } from './encodePointerFragment';
-
-export const parseWithPointers: SourceMapParser = <T>(value: string): IParserResult<T> => {
-  const parsed: IParserResult<T> = {
-    data: {} as T,
-    pointers: {},
-    validations: [],
-  };
-
-  if (!value || !value.trim().length) return parsed;
-
-  const validations: IValidationResult[] = [];
+export const parseWithPointers = <T = unknown>(value: string): IParserResult<T, IJsonASTNode, Map<number, number>> => {
```

Comment:
`T = unknown` is going to be annoying - suggest `any`. Most of the time we don't have a reliable typing to pass in to this generic in anyways, since we're dealing with user provided json.